homeassistant/helpers/system_info.py

```diff
@@ -1,25 +1,44 @@
 """Helper to gather system info."""
 from __future__ import annotations
 
+from functools import cache
 from getpass import getuser
+import logging
 import os
 import platform
 from typing import Any
 
 from homeassistant.const import __version__ as current_version
 from homeassistant.core import HomeAssistant
 from homeassistant.loader import bind_hass
-from homeassistant.util.package import is_virtual_env
+from homeassistant.util.package import is_docker_env, is_virtual_env
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@cache
+def is_official_image() -> bool:
+    """Return True if Home Assistant is running in an official container."""
+    return os.path.isfile("/OFFICIAL_IMAGE")
+
+
+# Cache the result of getuser() because it can call getpwuid() which
+# can do blocking I/O to look up the username in /etc/passwd.
+cached_get_user = cache(getuser)
 
 
 @bind_hass
 async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
     """Return info about the system."""
+    is_hassio = (
+        "hassio" in hass.config.components and hass.components.hassio.is_hassio()
+    )
```

Comment:
```suggestion
    is_hassio = hass.components.hassio.is_hassio()
```

Reply:
Won't that load hassio if it's not loaded without the guard ?

Reply:
Even default_config is doing `from homeassistant.components.hassio import is_hassio` but doesn't have `hassio` marked as a dep?

It works because there are so many places that unexpectedly load it.

It means that core and container systems always have the hassio integration loaded and it never does anything.

Reply:
This is a much larger issue and should get its own PR if we want to solve it

Reply:
`analytics`
`homeassistant_alerts`
`rtsp_to_webrtc`
`esphome`
`helpers/network.py`
`helpers/system_info`

The above all will load hassio on core/container systems